tools/gyp/gyp_main.py

```diff
@@ -6,10 +6,44 @@
 
 import os
 import sys
+import subprocess
+
+PY3 = bytes != str
+
+# Below IsCygwin() function copied from pylib/gyp/common.py
+def IsCygwin():
+  try:
+    out = subprocess.Popen("uname",
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT)
+    stdout, stderr = out.communicate()
+    if PY3:
+      stdout = stdout.decode("utf-8")
+    return "CYGWIN" in str(stdout)
```

Comment:
are we sure this can't be done with sth like `os.name` or `platform.system()`? Should I move this issue to `gyp3`?

Reply:
I don't know, Here's the PR: https://github.com/nodejs/node-gyp/pull/1817